test/helpers/framework.go

```diff
@@ -0,0 +1,298 @@
+			RequestURI(fmt.Sprintf("%s/instance", clusterLoggingURI)).
+			SetHeader("Content-Type", "application/json").
+			Do().Error()
+	})
+	logger.Debugf("Creating ClusterLogging: %s", string(body))
+	result := tc.KubeClient.RESTClient().Post().
+		RequestURI(clusterLoggingURI).
+		SetHeader("Content-Type", "application/json").
+		Body(body).
+		Do()
+	return result.Error()
+}
+
+func (tc *E2ETestFramework) Cleanup() {
+	logger.Debugf("Running %v e2e cleanup functions", len(tc.CleanupFns))
+	for _, cleanup := range tc.CleanupFns {
+		logger.Debug("Running an e2e cleanup function")
+		if err := cleanup(); err != nil {
+			logger.Debugf("Error during cleanup %v", err)
+		}
+	}
+}
+
+//newKubeClient returns a client using the KUBECONFIG env var or incluster settings
+func newKubeClient() (*kubernetes.Clientset, *rest.Config) {
+
+	var config *rest.Config
+	var err error
+	if kubeconfig := os.Getenv("KUBECONFIG"); kubeconfig != "" {
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		config, err = rest.InClusterConfig()
+	}
+	if err != nil {
+		panic(err.Error())
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+	return clientset, config
+}
+
+func (tc *E2ETestFramework) PodExec(namespace, name, container string, command []string) (string, error) {
+	req := tc.KubeClient.CoreV1().RESTClient().Post().
+		Namespace(namespace).
+		Resource("pods").
+		Name(name).
+		SubResource("exec").
+		Timeout(defaultTimeout).
+		VersionedParams(&corev1.PodExecOptions{
+			Container: container,
+			Command:   command,
+			Stdout:    true,
+			Stderr:    true,
+		}, scheme.ParameterCodec)
+
+	logger.Debugf("req: %v", req)
+	logger.Debugf("req.url: %v", req.URL().String())
+	exec, err := remotecommand.NewSPDYExecutor(tc.RestConfig, "POST", req.URL())
+	logger.Debugf("SPDY Error: %v", err)
+	if err != nil {
+		return "", err
+	}
+
+	var stdout, stderr bytes.Buffer
+	err = exec.Stream(remotecommand.StreamOptions{
+		Stdout: &stdout,
+		Stderr: &stderr,
+	})
+	logger.Debugf("exec.stream Error: %v", err)
+	logger.Debugf("stdout: %v", stdout.String())
+	logger.Debugf("stderr: %v", stderr.String())
+	if err != nil {
+		return "", err
+	}
+	return stdout.String(), nil
+}
+
+func (tc *E2ETestFramework) CreatePipelineSecret(pwd, logStoreName, secretName string) (secret *corev1.Secret, err error) {
+	logger.Debugf("Generating Pipeline certificates for %q", logStoreName)
+	if err = k8shandler.GenerateCertificates(OpenshiftLoggingNS, pwd, logStoreName, fmt.Sprintf("/tmp/clo-test-%d", rand.Intn(10000))); err != nil {
+		return nil, err
+	}
+	secret = k8shandler.NewSecret(
+		secretName,
+		OpenshiftLoggingNS,
+		map[string][]byte{
+			"tls.key":       utils.GetWorkingDirFileContents("logging-es.key"),
+			"tls.crt":       utils.GetWorkingDirFileContents("logging-es.crt"),
+			"ca-bundle.crt": utils.GetWorkingDirFileContents("ca.crt"),
+			"ca.key":        utils.GetWorkingDirFileContents("ca.key"),
+		},
+	)
```

Comment:
This secret is missing a cleanup; so after the e2e run it's a leftover

Reply:
added